<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1159</width>
    <height>776</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1110</width>
    <height>726</height>
   </size>
  </property>
  <property name="focusPolicy">
   <enum>Qt::StrongFocus</enum>
  </property>
  <property name="windowTitle">
   <string>Tetris3D</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget {
    background-color: #2d2d2d;
    color: #ffffff;
}

QPushButton {
    background-color: #555555;
    color: #ffffff;
    border: 1px solid #444444;
    border-radius: 5px;
    padding: 5px;
}

QPushButton:hover {
    background-color: #666666;
}

QPushButton:pressed {
    background-color: #333333; /* Цвет при нажатии */
    border: 1px solid #222222;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <widget class="QGraphicsView" name="graphicsView">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>800</width>
      <height>752</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>800</width>
      <height>700</height>
     </size>
    </property>
   </widget>
   <widget class="QGroupBox" name="scoreTable">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>211</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>15</pointsize>
      </font>
     </property>
     <property name="text">
      <string>BEST SCORE</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>67</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>current</string>
     </property>
    </widget>
    <widget class="QLabel" name="bestScore">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>10</y>
       <width>61</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <widget class="Line" name="line_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>191</width>
       <height>16</height>
      </rect>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="currentScore">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>70</y>
       <width>61</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
     </property>
    </widget>
    <zorder>bestScore</zorder>
    <zorder>currentScore</zorder>
    <zorder>label_12</zorder>
    <zorder>label_13</zorder>
    <zorder>line_3</zorder>
   </widget>
   <widget class="QGroupBox" name="settings">
    <property name="geometry">
     <rect>
      <x>819</x>
      <y>10</y>
      <width>331</width>
      <height>752</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>320</width>
      <height>700</height>
     </size>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QGroupBox" name="groupBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>311</width>
       <height>251</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>290</width>
       <height>250</height>
      </size>
     </property>
     <property name="title">
      <string>Block parameters</string>
     </property>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>291</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Number of approximating edges</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="roundingSlider">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>201</width>
        <height>31</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>40</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="rounding">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>149</y>
        <width>31</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>40</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::LinksAccessibleByMouse</set>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>190</y>
        <width>281</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>181</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Projection transparency</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>150</y>
        <width>21</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>%</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="blockTransparency">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>205</y>
        <width>71</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="minimum">
       <double>0.000000000000000</double>
      </property>
      <property name="maximum">
       <double>0.600000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>0.500000000000000</double>
      </property>
     </widget>
     <widget class="Line" name="line_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>281</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="approxEdges">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>60</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>13</pointsize>
       </font>
      </property>
      <property name="text">
       <string>2</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QSlider" name="approxEdgesSlider">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>60</y>
        <width>201</width>
        <height>21</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>20</number>
      </property>
      <property name="pageStep">
       <number>5</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>291</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Radius of rounding (as a percentage </string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="Line" name="line_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>281</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>291</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>of half the edge length of the cube)</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="Line" name="line_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>271</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>270</y>
       <width>301</width>
       <height>170</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>290</width>
       <height>170</height>
      </size>
     </property>
     <property name="title">
      <string>Game Field</string>
     </property>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>271</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Sizes</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QComboBox" name="depth">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>90</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>4</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>5</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>6</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="width">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>60</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>4</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>5</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>6</string>
       </property>
      </item>
     </widget>
     <widget class="QComboBox" name="height">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>120</y>
        <width>71</width>
        <height>25</height>
       </rect>
      </property>
      <property name="currentText">
       <string>7</string>
      </property>
      <item>
       <property name="text">
        <string>7</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>8</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>9</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>10</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>67</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>67</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Height</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>67</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Depth</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>454</y>
       <width>301</width>
       <height>171</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>171</width>
       <height>130</height>
      </size>
     </property>
     <property name="title">
      <string>Shadows and lighting</string>
     </property>
     <widget class="QCheckBox" name="withShadows">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>281</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>enable cast shadows</string>
      </property>
     </widget>
     <widget class="QComboBox" name="lightPos">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>281</width>
        <height>31</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Light from above from the side</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Light from the side</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>Light from above</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>281</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Position of the light source</string>
      </property>
     </widget>
    </widget>
    <widget class="QPushButton" name="startStopBtn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>700</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Start</string>
     </property>
    </widget>
    <widget class="QPushButton" name="newGameBtn">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>700</y>
       <width>141</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>New game</string>
     </property>
    </widget>
    <widget class="QPushButton" name="setSettingsBtn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>640</y>
       <width>301</width>
       <height>41</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="minimumSize">
      <size>
       <width>171</width>
       <height>41</height>
      </size>
     </property>
     <property name="text">
      <string>Apply settings</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
